### PR TITLE
Issue 4198

### PR DESCRIPTION
**Pull request by nbintertech**
connects #4198 

Add view factor calculation as link/button for user.

### CODE REVIEW

File: src/app/calculator/furnaces/opening/opening.service.ts [hunk 1]
```diff
@@ -208,6 +208,10 @@ export class OpeningService {
     };
   }
 
+    viewFactorCalculation(input: any, settings: Settings): number {
```
Comment (nbintertech) [inline src/app/calculator/furnaces/opening/opening.service.ts:211]: Wrapping this so I don't need to bring phastService into the form ts

File: src/app/calculator/furnaces/opening/opening-form/opening-form.component.ts [hunk 2]
```diff
@@ -141,6 +143,34 @@ export class OpeningFormComponent implements OnInit {
...
+    }
+  }
+
+  calculateViewFactor() {
+    this.calculate();
+    if (!this.canCalculateViewFactor) {
+      this.totalArea = 0.0;
+      return;
+    }
+    let vfInputs = this.openingService.getViewFactorInput(this.openingLossesForm);
```
Comment (rmroot) [inline src/app/calculator/furnaces/opening/opening-form/opening-form.component.ts:167]: add types

File: src/app/calculator/furnaces/opening/opening.service.ts (hunk 1 above)
Comment (rmroot) [reply src/app/calculator/furnaces/opening/opening.service.ts:211]: put a type on the inputs

File: src/app/phast/losses/opening-losses/opening-losses-form/opening-losses-form.component.ts [hunk 3]
```diff
@@ -127,7 +129,36 @@ export class OpeningLossesFormComponent implements OnInit {
...
+    }
+  }
+
+  calculateViewFactor() {
+    this.save();
+    if (!this.canCalculateViewFactor) {
+      this.totalArea = 0.0;
+      return;
+    }
+    let vfInputs = this.openingLossesService.getViewFactorInput(this.openingLossesForm);
```
Comment (rmroot) [inline src/app/phast/losses/opening-losses/opening-losses-form/opening-losses-form.component.ts:153]: types

**Review by rmroot — CHANGES_REQUESTED** [review]

**Review by rmroot — APPROVED** [review]

**Pull request merged**